Rate limit ioctl failures due to region database full messages

servesh opened the pull request:
On Aurora when using kdreg2 the logs are flooded with region database full messages. This eventually causes our backend log aggregation system to be overwhelmed. So proposing to ratelimit these functions.

If there are other parts of the code such behavior is expected, would be good to ratelimit them as well.

d314159 committed:
NETCASSINI-7012: enhance shs dkms builds to allow parallelism

Signed-off-by: dennis-c-josifovich <dennis.c.josifovich@hpe.com>

jswaro committed:
NETCASSINI-7164: Create directory if necessary (#80)

If the module load directry does not exist, then
the rpm post scriptlet will fail. Create the directory.

Signed-off-by: James Swaro <james.swaro@hpe.com>

servesh committed:
Rate limit ioctl failures due to region database full messages

Signed-off-by: dennis-c-josifovich <dennis.c.josifovich@hpe.com>

servesh commented:
Sample messages being seen:
```
[Fri Mar 21 01:32:07 2025] [156430] kdreg2:kdreg2_monitor_region: Region database full, rejecting request to monitor region.
[Fri Mar 21 01:32:07 2025] [156426] kdreg2:kdreg2_monitor_region: Region database full, rejecting request to monitor region.
[Fri Mar 21 01:32:07 2025] [156430] kdreg2:kdreg2_ioctl: KDREG2_IOCTL_MONITOR: failure -28
[Fri Mar 21 01:32:07 2025] [156426] kdreg2:kdreg2_ioctl: KDREG2_IOCTL_MONITOR: failure -28
[Fri Mar 21 01:32:07 2025] [156433] kdreg2:kdreg2_monitor_region: Region database full, rejecting request to monitor region.
[Fri Mar 21 01:32:07 2025] [156433] kdreg2:kdreg2_ioctl: KDREG2_IOCTL_MONITOR: failure -28
[Fri Mar 21 01:32:07 2025] [156430] kdreg2:kdreg2_monitor_region: Region database full, rejecting request to monitor region.
[Fri Mar 21 01:32:07 2025] [156430] kdreg2:kdreg2_ioctl: KDREG2_IOCTL_MONITOR: failure -28
[Fri Mar 21 01:32:07 2025] [156433] kdreg2:kdreg2_monitor_region: Region database full, rejecting request to monitor region.
[Fri Mar 21 01:32:07 2025] [156433] kdreg2:kdreg2_ioctl: KDREG2_IOCTL_MONITOR: failure -28
```

servesh commented:
Additional messages that are seen that needs to be rate limited as well,
```
kdreg2:kdreg2_context_resize: resize to 1024 entities
kdreg2:kdreg2_open: Instance opened.
kdreg2:kdreg2_detect_fork: Fork() detected - monitoring not supported in child
kdreg2:kdreg2_release: Instance closed.
```

iziemba commented:
Hi Servesh. Thanks for opening this. HPE is working on incorporating in on the main branch. Once landed on main, it will be cherry-picked to release/shs-12.0.

jswaro committed:
pipeline: Remove the obsolete pipeline steps (#84)

cleanWs was removed, and checkout can be performed at a higher
level within Jenkins. Remove it from the pipeline

Signed-off-by: James Swaro <james.swaro@hpe.com>

servesh committed:
Rate limit more messages seen while testing on Aurora

servesh commented:
Noticed that the main branch didn't have further changes noted above in comments. So updated the branch with the new commit that addresses further messages identified on Aurora.
